extensions/binding/org.eclipse.smarthome.binding.openweathermap/src/main/java/org/eclipse/smarthome/binding/openweathermap/internal/discovery/OpenWeatherMapLocationDiscoveryService.java

```diff
@@ -0,0 +1,172 @@
+        this.bridgeHandler = bridgeHandler;
+        this.locationProvider = locationProvider;
+        this.localeProvider = localeProvider;
+        this.i18nProvider = i18nProvider;
+        activate(null);
+    }
+
+    @Activate
+    @Override
+    protected void activate(@Nullable Map<String, @Nullable Object> configProperties) {
+        super.activate(configProperties);
+    }
+
+    @Modified
+    @Override
+    protected void modified(@Nullable Map<String, @Nullable Object> configProperties) {
+        super.modified(configProperties);
+    }
+
+    @Deactivate
+    @Override
+    public void deactivate() {
+        removeOlderResults(new Date().getTime(), bridgeHandler.getThing().getUID());
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Start manual OpenWeatherMap Location discovery scan.");
+        scanForNewLocation();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        logger.debug("Stop manual OpenWeatherMap Location discovery scan.");
+        super.stopScan();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            logger.debug("Start OpenWeatherMap Location background discovery job at intervall {} s.",
+                    DISCOVERY_INTERVAL_SECONDS);
+            discoveryJob = scheduler.scheduleWithFixedDelay(this::scanForNewLocation, 0, DISCOVERY_INTERVAL_SECONDS,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        if (discoveryJob != null && !discoveryJob.isCancelled()) {
+            logger.debug("Stop OpenWeatherMap Location background discovery job.");
+            if (discoveryJob.cancel(true)) {
+                discoveryJob = null;
+            }
+        }
+    }
+
+    private void scanForNewLocation() {
+        PointType currentLocation = locationProvider.getLocation();
+        if (currentLocation == null) {
+            logger.debug("Location is not set -> Will not provide any discovery results.");
+        } else if (!Objects.equals(currentLocation, previousLocation)) {
+            logger.debug("Location has been changed from {} to {} -> Creating new discovery results.", previousLocation,
+                    currentLocation);
+            createResults(currentLocation);
+            previousLocation = currentLocation;
+        } else {
+            createResults(currentLocation);
+        }
+    }
+
+    private void createResults(PointType location) {
+        String locationString = location.toFullString();
+        ThingUID bridgeUID = bridgeHandler.getThing().getUID();
+        createWeatherResult(locationString, bridgeUID);
+        OpenWeatherMapAPIConfiguration config = bridgeHandler.getOpenWeatherMapAPIConfig();
+        if (config.getHourlyForecast() == Boolean.TRUE) {
```

Comment:
maybe I understand now how those config params work: in the bridge config the user can decide if hourly and/or daily forecasts things should be discovered. In case one option is `false` auto discovery and data fetching get disabled.
But: when the daily forecast thing gets discovered, isn't it an option to just ignore the discovery result and decide data fetching by the existing things on the bridge?

Reply:
@htreu Thank you for taking time to review the binding. Yes, I agree. It makes more sense. I cannot remember why I did it this was. Maybe I should rename those parameters to `discoverHourlyForecast` and `discoverDailyForecast`. The purpose should be clearer then.

Reply:
See my comment on the description for both config parameters.

Reply:
Done.